### PR TITLE
Implement fetching GitHub repositories

### PR DESCRIPTION
**Pull request by satococoa**
This pull request introduces GitHub integration into the playground application.
The integration includes fetching repositories, handling user authorization, and updating the UI to reflect the integration status.
The most important changes are grouped into context setup, provider implementation, and UI updates.

### Context and Provider Setup:
* Added `GitHubIntegrationContext` and `useGitHubIntegration` hook to manage GitHub integration state. (`app/(playground)/p/[agentId]/beta-proto/github-integration/context.ts`) ([app/(playground)/p/[agentId]/beta-proto/github-integration/context.tsR1-R24](diffhunk://#diff-7373575e6660c7725a8fe9023e2e0cdc8ec5659224da973095e08542aa66b234R1-R24))
* Implemented `GitHubIntegrationProvider` to supply the GitHub integration state to the component tree. (`app/(playground)/p/[agentId]/beta-proto/github-integration/provider.tsx`) ([app/(playground)/p/[agentId]/beta-proto/github-integration/provider.tsxR1-R23](diffhunk://#diff-03183a27524ce714a9f90ee38bde3890b4df99b87523c1e416d814f1137dbd76R1-R23))

### UI Updates:
* Updated `PlaygroundProps` and `Playground` component to include GitHub integration. (`app/(playground)/p/[agentId]/beta-proto/component.tsx`) ([app/(playground)/p/[agentId]/beta-proto/component.tsxR29-R44](diffhunk://#diff-bc7a83854876f570ecd8ec885abb9743562708e86ad753c89ee96034d736bdd4R29-R44))
* Modified `GitHubIntegration` component to use real repository data and handle authorization status. (`app/(playground)/p/[agentId]/beta-proto/left-menu/github-integration/github-integration.tsx`) ([app/(playground)/p/[agentId]/beta-proto/left-menu/github-integration/github-integration.tsxR13-L31](diffhunk://#diff-7b27c13a2e68170a9f12ae84244f925787f07b3cf2dfc009f10301e3ce84557aR13-L31), [app/(playground)/p/[agentId]/beta-proto/left-menu/github-integration/github-integration.tsxR55-L82](diffhunk://#diff-7b27c13a2e68170a9f12ae84244f925787f07b3cf2dfc009f10301e3ce84557aR55-L82), [app/(playground)/p/[agentId]/beta-proto/left-menu/github-integration/github-integration.tsxL91-R85](diffhunk://#diff-7b27c13a2e68170a9f12ae84244f925787f07b3cf2dfc009f10301e3ce84557aL91-R85), [app/(playground)/p/[agentId]/beta-proto/left-menu/github-integration/github-integration.tsxR159-R178](diffhunk://#diff-7b27c13a2e68170a9f12ae84244f925787f07b3cf2dfc009f10301e3ce84557aR159-R178))
* Integrated GitHub repository fetching into the `AgentPlaygroundPage` component. (`app/(playground)/p/[agentId]/page.tsx`) ([app/(playground)/p/[agentId]/page.tsxR80-R113](diffhunk://#diff-76ad14a1cddab8e31f27e0f060bd5759b440c88db08545922f3e7b8281856437R80-R113), [app/(playground)/p/[agentId]/page.tsxR137](diffhunk://#diff-76ad14a1cddab8e31f27e0f060bd5759b440c88db08545922f3e7b8281856437R137), [app/(playground)/p/[agentId]/page.tsxR153-R156](diffhunk://#diff-76ad14a1cddab8e31f27e0f060bd5759b440c88db08545922f3e7b8281856437R153-R156))

### CODE REVIEW

File: app/(playground)/p/[agentId]/beta-proto/left-menu/github-integration/github-integration.tsx
Comment (toyamarinyon) [inline app/(playground)/p/[agentId]/beta-proto/left-menu/github-integration/github-integration.tsx:1]: I think that extracting `goToAccountSettings` and `githubIntegrationForm` into separate components enhances code maintainability and follows the Single Responsibility Principle. 

By creating dedicated GoToAccountSettings and GitHubIntegrationForm components, each piece of code serves a clear purpose - managing authorization state and handling integration configuration.

This separation simplifies bug fixes, enables independent testing, and makes components reusable across the application. The improved organization leads to better team collaboration and easier code reviews, ultimately creating a more efficient and scalable codebase. 

```tsx
// ... original codes
function GoToAccountSettings() {
    return (
        <div className="grid gap-[16px]">
            <div className="text-sm text-gray-600">
                Please connect your GitHub account to get started
            </div>
            <Button asChild>
                <Link href="/settings/account">Connect GitHub</Link>
            </Button>
        </div>
    );
}

interface GitHubIntegrationFormProps {
    repositories: Array<{
        id: number;
        full_name: string;
    }>;
}

function GitHubIntegrationForm({ repositories }: GitHubIntegrationFormProps) {
    return (
        <div className="grid gap-[16px]">
            <Section>
                <SectionHeader title="Repository" />
                <Select>
                    <SelectTrigger>
                        <SelectValue placeholder="Choose value" />
                    </SelectTrigger>
                    <SelectContent>
                        {repositories.map((repository) => (
                            <SelectItem
                                value={repository.id.toString()}
                                key={repository.id}
                            >
                                {repository.full_name}
                            </SelectItem>
                        ))}
                    </SelectContent>
                </Select>
            </Section>
            <Section>
                <SectionHeader title="Trigger" />
                <SectionFormField>
                    <Label htmlFor="event">Event</Label>
                    <Select name="event">
                        <SelectTrigger>
                            <SelectValue placeholder="Choose value" />
                        </SelectTrigger>
                        <SelectContent>
                            {mockEvents.map((event) => (
                                <SelectItem value={event.id} key={event.id}>
                                    {event.name}
                                </SelectItem>
                            ))}
                        </SelectContent>
                    </Select>
                </SectionFormField>
                <SectionFormField>
                    <Label htmlFor="command">Call sign</Label>
                    <Input
                        type="text"
                        name="command"
                        placeholder="Enter call sign"
                        className="w-full"
                    />
                    <span className="text-black-70 text-[12px]">
                        You can call this agent by commenting{" "}
                        <span className="py-[0px] px-[4px] text-black--30 bg-black-70 rounded-[2px]">
                            /giselle report-agent
                        </span>{" "}
                        in the issue route06inc/giselle.
                    </span>
                </SectionFormField>
            </Section>
            <Section>
                <SectionHeader title="Action" />
                <SectionFormField>
                    <Label htmlFor="event">Run flow</Label>
                    <Select name="event">
                        <SelectTrigger>
                            <SelectValue placeholder="Choose value" />
                        </SelectTrigger>
                        <SelectContent>
                            {mockNodes.map((node) => (
                                <SelectItem value={node.id} key={node.id}>
                                    {node.name}
                                </SelectItem>
                            ))}
                        </SelectContent>
                    </Select>
                </SectionFormField>
                <SectionFormField>
                    <Label htmlFor="event">Then</Label>
                    <Select name="event">
                        <SelectTrigger>
                            <SelectValue placeholder="Choose value" />
                        </SelectTrigger>
                        <SelectContent>
                            {mockNextActions.map((nextAction) => (
                                <SelectItem value={nextAction.id} key={nextAction.id}>
                                    {nextAction.name}
                                </SelectItem>
                            ))}
                        </SelectContent>
                    </Select>
                </SectionFormField>
            </Section>
        </div>
    );
}

export function GitHubIntegration(props: GitHubIntegrationProps) {
    const { needsAuthorization, repositories } = useGitHubIntegration();

    return (
        <div className="grid gap-[24px] px-[24px] py-[24px]">
            <header className="flex justify-between">
                <p
                    className="text-[22px] font-rosart text-black--30"
                    style={{ textShadow: "0px 0px 20px hsla(207, 100%, 48%, 1)" }}
                >
                    GitHub Integration
                </p>
                <button type="button">
                    <XIcon
                        className="w-[16px] h-[16px]"
                        onClick={() => props.setTabValue("")}
                    />
                </button>
            </header>
            {needsAuthorization ? (
                <GoToAccountSettings />
            ) : (
                <GitHubIntegrationForm repositories={repositories} />
            )}
        </div>
    );
}
```

**Review by toyamarinyon — COMMENTED** [review]
Thanks! It is generally OK. If you write a component within a component, you will lose visibility of the code, so just adjust there.

**Review by shige — APPROVED** [review]
LGTM 👍

File: app/(playground)/p/[agentId]/beta-proto/left-menu/github-integration/github-integration.tsx
Comment (satococoa) [reply app/(playground)/p/[agentId]/beta-proto/left-menu/github-integration/github-integration.tsx:1]: Thanks for the feedback! I'll make those changes 😄

**Review by toyamarinyon — APPROVED** [review]
Thanks!

**Pull request merged**